src/iidm/util/SV.cpp

```diff
@@ -0,0 +1,191 @@
+    return computeU2(y, std::complex<double>(0, 0), z, rho);
+}
+
+double SV::getA() const {
+    return m_a;
+}
+
+double SV::getB(const TwoWindingsTransformer& twt) {
+    return twt.getB() * (twt.hasRatioTapChanger() ? 1 + twt.getRatioTapChanger().getCurrentStep().getB() / 100 : 1.0);
+}
+
+double SV::getG(const TwoWindingsTransformer& twt) {
+    return twt.getG() * (twt.hasRatioTapChanger() ? 1 + twt.getRatioTapChanger().getCurrentStep().getG() / 100 : 1.0);
+}
+
+double SV::getP() const {
+    return m_p;
+}
+
+double SV::getQ() const {
+    return m_q;
+}
+
+double SV::getR(const TwoWindingsTransformer& twt) {
+    return twt.getR() * (twt.hasRatioTapChanger() ? 1 + twt.getRatioTapChanger().getCurrentStep().getR() / 100 : 1.0);
+}
+
+double SV::getRho(const TwoWindingsTransformer& twt) {
+    return twt.getRatedU2() / twt.getRatedU1() * (twt.hasRatioTapChanger() ? 1 + twt.getRatioTapChanger().getCurrentStep().getRho() / 100 : 1.0);
+}
+
+double SV::getU() const {
+    return m_u;
+}
+
+double SV::getX(const TwoWindingsTransformer& twt) {
+    return twt.getX() * (twt.hasRatioTapChanger() ? 1 + twt.getRatioTapChanger().getCurrentStep().getX() / 100 : 1.0);
+}
+
+SV SV::otherSide(double r, double x, double g, double b, double rho) const {
+    std::complex<double> z(r, x);
+    std::complex<double> y(g, b);
+    std::complex<double> u2 = computeU2(y, z, rho);
+    std::complex<double> s2 = computeS2(y, z, rho);
+    return {-s2.real(), -s2.imag(), std::abs(u2), std::arg(u2) * stdcxx::toDegrees};
+}
+
+SV SV::otherSide(double r, double x, double g1, double b1, double g2, double b2, double rho) const {
+    std::complex<double> z (r, x);
+    std::complex<double> y1(g1, b1);
+    std::complex<double> y2(g2, b2);
+    std::complex<double> u2 = computeU2(y1, y2, z, rho);
+    std::complex<double> s2 = computeS2(y1, y2, z, rho);
+    return {-s2.real(), -s2.imag(), std::abs(u2), std::arg(u2) * stdcxx::toDegrees};
+}
+
+SV SV::otherSide(const TwoWindingsTransformer& twt) const {
+    return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), getRho(twt));
+}
+
+SV SV::otherSide(const Line& line) const {
+    return otherSide(line.getR(), line.getX(), line.getG1() + line.getG2(), line.getB1() + line.getB2(), 1);
+}
+
+SV SV::otherSide(const DanglingLine& danglingLine) const {
+    return otherSide(danglingLine.getR(), danglingLine.getX(), danglingLine.getG() / 2.0, danglingLine.getB() / 2.0, danglingLine.getG() / 2.0, danglingLine.getB() / 2.0, 1);
+}
+
+double SV::otherSideA(double r, double x, double g1, double b1, double rho) const {
+    std::complex<double> z(r, x);
+    std::complex<double> y1(g1, b1);
+    std::complex<double> u2 = computeU2(y1, std::complex<double>(), z, rho);
+    return std::arg(u2) * stdcxx::toDegrees;
+}
+
+double SV::otherSideA(const TieLine::HalfLine& halfLine) const {
+    return otherSideA(halfLine.getR(), halfLine.getX(), halfLine.getG1(), halfLine.getB1(), 1.0);
```

Comment:
I also have a doubt here: the other side of the HalfLine should take into account `g2` and `b2` (@zamarrenolm) to be consistent with `otherSideP` and `otherSideQ`